spec/javascripts/customJestMatchers.ts

```diff
@@ -0,0 +1,27 @@
+import { expect } from '@jest/globals'
+import {
+  printExpected,
+  printReceived
+} from 'jest-matcher-utils'
+
+const toBeWithinSecondsFrom = (actual: Date, expected: Date, seconds = 1) => {
+  const diff = Math.abs(expected.getTime() - actual.getTime())
+  const pass = diff <= seconds * 1000
+  const message = () => `expected ${printReceived(actual)} ${pass ? 'not ' : ''}to be within ${seconds} seconds from ${printExpected(expected)}, but it was within ${diff / 1000} seconds`
```

Comment:
I don't understand this part: `${pass ? 'not ' : ''}`. The message will only be shown when `pass == false`, so why this?

Reply:
The message will be shown also when we `expect(value).not.toBeWithinSecondsFrom(targetDate)` (note the **not**), but the check passes. In this case we need to add a negation in the error message, e.g.
```
expected 2025-05-19T12:17:50.595Z not to be within 1 seconds from 2025-05-19T12:17:50.646Z, but it was within 0.051 seconds
```

Reply:
Ah I see! thanks for clarifying